_posts/2019-07-30-censored-data/censored_data_tfp.Rmd

```diff
@@ -0,0 +1,724 @@
+As well as for `ns_import`, the estimated number of imported functions or methods:
+
+::: l-body
+| percentiles: _ns_import_    |   10%   |   30%   |   50%   |   70%   |   90%  |    100%   |
+|-----------------------------|---------|---------|---------|---------|--------|-----------|
+| completed                   |     0   |     1   |     3   |     6   |    19  |     312   |
+| not completed               |     0   |     2   |     5   |    11   |    23  |     297   |
+:::
+
+Same pattern for `r_size`, the size on disk of files in the `R` directory:
+
+::: l-body
+| percentiles: _r_size_       |   10%   |   30%   |   50%   |   70%   |   90%  |    100%   |
+|-----------------------------|---------|---------|---------|---------|--------|-----------|
+| completed                   |   0.005 |   0.015 |   0.031 |   0.063 |  0.176 |     3.746 |
+| not completed               |   0.008 |   0.019 |   0.041 |   0.097 |  0.217 |     2.148 |
+:::
+
+And finally, we see it for `doc_size` too, where `doc_size` is the size of `.Rmd` and `.Rnw` files:
+
+::: l-body
+| percentiles: _doc_size_     |   10%   |   30%   |   50%   |   70%   |   90%  |    100%   |
+|-----------------------------|---------|---------|---------|---------|--------|-----------|
+| completed                   |   0.000 |   0.000 |   0.000 |   0.000 |  0.023 |     0.988 |
+| not completed               |   0.000 |   0.000 |   0.000 |   0.011 |  0.042 |     0.114 |
+:::
+
+Given our task at hand -- model check durations taking into account uncensored as well as censored data -- we won't dwell on differences between both groups any longer; nonetheless we thought it interesting to relate these numbers.
+
+So now, back to work. We need to create a model.
+
+## The model
+
+As explained in the introduction, for completed checks duration is modeled using an exponential PDF. This is as straightforward as adding [tfd_exponential()](https://rstudio.github.io/tfprobability/reference/tfd_exponential.html) to the model function, [tfd_joint_distribution_sequential()](https://rstudio.github.io/tfprobability/reference/tfd_joint_distribution_sequential.html). ^[For a first introduction to MCMC sampling with `tfprobability`, see [Tadpoles on TensorFlow: Hierarchical partial pooling with tfprobability](https://blogs.rstudio.com/tensorflow/posts/2019-05-06-tadpoles-on-tensorflow/)]For the censored portion, we need the exponential CCDF. This one is not, as of today, easily added to the model. What we can do though is calculate its value ourselves and add it to the "main" model likelihood. We'll see this below when discussing sampling; for now it means the model definition ends up straightforward as it only covers the non-censored data. It is made of just the said exponential PDF and priors for the regression parameters.
+
+As to the latter, we use 0-centered, wide, Gaussian priors for all parameters. As the priors are all the same, instead of listing a bunch of `tfd_normal`s, we can create them all at once as
```

Comment:
```suggestion
As for the latter, we use 0-centered, wide, Gaussian priors for all parameters. As the priors are all the same, instead of listing a bunch of `tfd_normal`s, we can create them all at once as
```